### DISCUSSION TITLE
SES overrides don't work as expected

### DISCUSSION THREAD

**Discussion by then3rdman** [magicbell/community#64] [Questions] [answered]
It seems that the channel overides do nothing. At least for the email channel. I will also point out that the API documentation is different from that of the specific docs page.

One has the overrides nested under `channels` and the other does not. I have tried both and neither work. 

Additionally the TS type for `NewNotification` is completely missing the `overrides` field.

I have also seen another bug report regarding the overrides for `email.acttion_url` is not working. This is over 2 months old....

To be very honest, a paid product with what is essentially a non funcitonal features for multiple months is a pretty big deal.

**Comment by unamashana** [comment]
Hey @then3rdman I am very sorry you are having this experience with MagicBell. We need to pick up pace with some of these issues. There is a known issue with the email channel overrides not making their way through. 

However, the provider specific overrides for email work. Which email provider have you tried them for? We are happy to help with them.

Where are you doing the email rendering? 

If it's in the third party, you should be able to use custom_attributes there - we relay them. Therefore you can override the URLs. If you are doing it in your code, you can simply send us the rendered HTML and skip other overrides all together. We can also relay this rendered HTML to the third party providers. 

We have been slower on bugs with workarounds, and we are happy to help you find a solution. Several of our customers are using emails very successfully, and we can help you do so too. Thanks.

**Comment by then3rdman** [comment]
Thanks for the response. Sorry for the aggressive tone. So I have managed to resolve the overide of content not working howver unexpectedly it is completely bypassing the email template, both global and channel specific

**Reply by unamashana** [reply]
No problem - everything you mentioned is valid and a challenge if you don't know the workarounds. We appreciate the candid feedback. 

Also, as I noted below, the content override is designed to give you full control on rendering. Therefore we do not override it with any other input.

**Comment by then3rdman** [comment]
So currently the `content` override means that the email is delivered with no template and the `action_url` override has no effect. Currently both are not usuable

**Comment by unamashana** [comment]
Hey @then3rdman `content` override is designed that way. I am glad it's working for you.

**Reply by then3rdman** [reply]
OK, does seems to odd to hard bypass two levels of templating with just a string input.

**Comment by baltamar3** [comment]
Hi, I have the same problem with `overrides.providers.ses.Source` (Not working)

**Reply by unamashana** [reply]
Hey @baltamar3 I am sorry for the trouble. We checked, and the code is setting up overrides. Do you see it in the request payload to SES in our Log page, for email deliveries? Can I also request you to check SES logs? While it's possible that we are not setting the overrides, it's also possible that AWS is not using what you set, because the email is not a verified sender, or for other AWS reason.

**Reply by baltamar3** [reply]
- notification creation endpoint, changing amazon ses source to **`Subrik Notifications <no-reply@subrik.com>`**
![image](https://user-images.githubusercontent.com/20992846/232107180-a0c71fe6-d6f8-4bf5-82e7-170ad496ece4.png)

- source override is not reflected in magicbell logs
![image](https://user-images.githubusercontent.com/20992846/232107542-d6cbb470-45ac-455c-8cc2-f34473d7d47e.png)

**Comment by gamer22026** [comment]
Ditto for me with SES. Overrides for anything (including the example of Source) simply do not work.

**Reply by unamashana** [reply]
Sorry about that - we will work on reproducing and fixing this. Thanks!

**Comment by unamashana** [comment, marked as answer]
We are very sorry for the confusion. The override key was incorrect in the docs. The correct key to use is `notification.overrides.providers.amazon_ses`.  For more info, please refer to the [SES page on MagicBell docs](https://www.magicbell.com/docs/email/amazon-ses)
